File: equipment-failure-prediction/backend/src/main/java/com/suyog/equipmentfailureprediction/service/PredictionService.java
package com.suyog.equipmentfailureprediction.service;

import com.suyog.equipmentfailureprediction.dto.request.PredictionRequest;
import com.suyog.equipmentfailureprediction.dto.response.PredictionResponse;

public interface PredictionService {

    PredictionResponse predict(PredictionRequest request);

}

File: equipment-failure-prediction/frontend/src/services/predictionService.js
import api from './api';

export const predictionService = {
  predict: async (data) => {
    // Expects: { equipmentId, airTemperature, processTemperature, rotationalSpeed, torque, toolWear }
    const response = await api.post('/predictions', data);
    return response.data; // PredictionResponse directly (not wrapped in ApiResponse)
  },

  uploadCsv: async (file) => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await api.post('/predictions/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data; // CsvPredictionResponse directly (not wrapped in ApiResponse)
  },
};

export default predictionService;
